frame/block_builder/block_builder.go

```diff
@@ -12,51 +22,197 @@ type BlockBuilder interface {
-*/
-func FinalizeBlock(dataPtr int32, dataLen int32) int64
+	systemHash := hashing.Twox128(constants.KeySystem)
+	numberHash := hashing.Twox128(constants.KeyNumber)
+
+	bNumber := storage.Get(append(systemHash, numberHash...))
+	buf := &bytes.Buffer{}
+	buf.Write(bNumber)
+	blockNumber := goscale.DecodeU32(buf)
+
+	idleAndFinalizeHook(types.BlockNumber{U32: blockNumber})
+
+	header := finalize()
+	encodedHeader := header.Bytes()
+
+	return utils.BytesToOffsetAndSize(encodedHeader)
+}
 
 /*
-	https://spec.polkadot.network/#defn-rt-builder-inherent-extrinsics
+https://spec.polkadot.network/#defn-rt-builder-inherent-extrinsics
 
-	SCALE encoded arguments (data types.InherentsData) allocated in the Wasm VM memory, passed as:
-		dataPtr - i32 pointer to the memory location.
-		dataLen - i32 length (in bytes) of the encoded arguments.
-		returns a pointer-size to the SCALE-encoded ([]types.Extrinsic) data.
+SCALE encoded arguments (data types.InherentsData) allocated in the Wasm VM memory, passed as:
+
+	dataPtr - i32 pointer to the memory location.
+	dataLen - i32 length (in bytes) of the encoded arguments.
+	returns a pointer-size to the SCALE-encoded ([]types.Extrinsic) data.
 */
-func InherentExtrinisics(dataPtr int32, dataLen int32) int64
+func InherentExtrinisics(dataPtr int32, dataLen int32) int64 { return 0 }
 
 /*
-	https://spec.polkadot.network/#id-blockbuilder_check_inherents
+https://spec.polkadot.network/#id-blockbuilder_check_inherents
+
+SCALE encoded arguments (block types.Block, data types.InherentsData) allocated in the Wasm VM memory, passed as:
 
-	SCALE encoded arguments (block types.Block, data types.InherentsData) allocated in the Wasm VM memory, passed as:
-		dataPtr - i32 pointer to the memory location.
-		dataLen - i32 length (in bytes) of the encoded arguments.
-		returns a pointer-size to the SCALE-encoded ([]byte) data.
+	dataPtr - i32 pointer to the memory location.
+	dataLen - i32 length (in bytes) of the encoded arguments.
+	returns a pointer-size to the SCALE-encoded ([]byte) data.
 */
-func CheckInherents(dataPtr int32, dataLen int32) int64
+func CheckInherents(dataPtr int32, dataLen int32) int64 {
+	return 0
+}
 
 /*
-	https://spec.polkadot.network/#id-blockbuilder_random_seed
+https://spec.polkadot.network/#id-blockbuilder_random_seed
 
-	SCALE encoded arguments () allocated in the Wasm VM memory, passed as:
-		dataPtr - i32 pointer to the memory location.
-		dataLen - i32 length (in bytes) of the encoded arguments.
-		returns a pointer-size to the SCALE-encoded ([32]byte) data.
+SCALE encoded arguments () allocated in the Wasm VM memory, passed as:
+
+	dataPtr - i32 pointer to the memory location.
+	dataLen - i32 length (in bytes) of the encoded arguments.
+	returns a pointer-size to the SCALE-encoded ([32]byte) data.
 */
-func RandomSeed(dataPtr int32, dataLen int32) int64
+func RandomSeed(dataPtr int32, dataLen int32) int64 { return 0 }
+
+func noteFinishedExtrinsics() {
```

Comment:
`frame/system/system.go` could be a better place for `noteFinishedExtrinsics` since it's main job is to interact with the storage (and also will match the substrate's implementation)